### PR TITLE
✨ Improved time zone selector

### PR DESCRIPTION
**Pull request by lukevella**
## Summary by CodeRabbit

* **New Features**
  * Inline, searchable timezone picker showing city names and UTC offsets (replaces modal selector)
  * Auto-generated, curated timezone dataset powering the picker
  * New combobox and input-group UI primitives for improved form interactions

* **Tests**
  * Comprehensive unit tests for timezone selection: rendering, filtering, selection, and edge cases

* **Chores**
  * Centralized timezone validation/canonicalization for consistent handling across the app
  * Minor UI token: global radius CSS variable added

### CODE REVIEW

File: apps/web/src/utils/date-time-utils.ts [hunk 1]
```diff
@@ -1,85 +1,12 @@
...
-  }
-
-  if (!tz) {
-    // In theory this shouldn't happen
-    tz = "America/New_York";
-    console.error({ timeZone }, "Unable to resolve timezone");
-  }
-
-  return tz;
+  return dayjs.tz.guess();
```
[the remark this answers is not in the document]
Comment (lukevella) [reply apps/web/src/utils/date-time-utils.ts:9]: The app should still work if a user's timezone resolves to be a fixed offset however when they set their timezone we expect them to choose from a list of geographic timezones so they should never be able to send a fixed offset timezone to the server.
